plugins/inputs/gtfs/gtfs.go

```diff
@@ -0,0 +1,305 @@
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type GTFS struct {
+	Key                 string `toml:"key"`
+	VehiclePositionsURL string `toml:"vehicle_positions_url"`
+	TripUpdatesURL      string `toml:"trip_updates_url"`
+	ServiceAlertsURL    string `toml:"service_alerts_url"`
+
+	Username string            `toml:"username"`
+	Password string            `toml:"password"`
+	Timeout  internal.Duration `toml:"timeout"`
+
+	tls.ClientConfig
+
+	client              *http.Client
+	vehiclePositionsReq *http.Request
+	tripUpdatesReq      *http.Request
+	serviceAlertsReq    *http.Request
+}
+
+func (s *GTFS) Description() string {
+	return "GTFS-realtime data"
+}
+
+func (s *GTFS) SampleConfig() string {
+	return `
+  ## API Key
+  # key = "${GTFS_API_KEY}"
+  ## URL for fetching vehicle positions
+  # vehicle_positions_url = "https://host.test/VehiclePositions.pb"
+  ## URL for fetching vehicle positions
+  # trip_updates_url = "https://host.test/TripUpdates.pb"
+  ## URL for fetching vehicle positions
+  # service_alerts_url = "https://host.test/ServiceAlerts.pb"
+
+  ## Optional HTTP Basic Auth Credentials
+  # username = "username"
+  # password = "pa$$word"
+
+  ## Optional TLS Config
+  # tls_ca = "/etc/telegraf/ca.pem"
+  # tls_cert = "/etc/telegraf/cert.pem"
+  # tls_key = "/etc/telegraf/key.pem"
+  ## Use TLS but skip chain & host verification
+  # insecure_skip_verify = false
+
+  ## Amount of testTimestamp allowed to complete the HTTP request
+  # timeout = "5s"
+`
+}
+
+func (g *GTFS) Init() error {
+	tlsCfg, err := g.ClientConfig.TLSConfig()
+	if err != nil {
+		return err
+	}
+
+	g.client = &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: tlsCfg,
+			Proxy:           http.ProxyFromEnvironment,
+		},
+		Timeout: g.Timeout.Duration,
+	}
+
+	if g.VehiclePositionsURL == "" && g.TripUpdatesURL == "" && g.ServiceAlertsURL == "" {
+		return errNoSourceConfigured
+	}
+
+	if g.VehiclePositionsURL != "" {
+		r, err := g.newRequest(g.VehiclePositionsURL)
+		if err != nil {
+			return err
+		}
+		g.vehiclePositionsReq = r
+	}
+
+	if g.TripUpdatesURL != "" {
+		r, err := g.newRequest(g.TripUpdatesURL)
+		if err != nil {
+			return err
+		}
+		g.tripUpdatesReq = r
+	}
+
+	if g.ServiceAlertsURL != "" {
+		r, err := g.newRequest(g.ServiceAlertsURL)
+		if err != nil {
+			return err
+		}
+		g.serviceAlertsReq = r
+	}
+
+	return nil
+}
+
+func (g *GTFS) Gather(acc telegraf.Accumulator) error {
+	var (
+		work errgroup.Group
+		now  = time.Now()
+	)
+
+	if g.vehiclePositionsReq != nil {
+		work.Go(func() error { return g.gatherVehiclePositions(acc, now) })
+	}
+
+	if g.tripUpdatesReq != nil {
+		work.Go(func() error { return g.gatherTripUpdates(acc, now) })
+	}
+
+	if g.serviceAlertsReq != nil {
+		work.Go(func() error { return g.gatherServiceAlerts(acc, now) })
+	}
+
+	return work.Wait()
+}
+
+func (g *GTFS) gatherVehiclePositions(acc telegraf.Accumulator, t time.Time) error {
```

Comment:
I don't see where you use `t`. Please remove the parameter.

Reply:
I believe this is an oversight and it should be used as the timestamp. Will double check and use or remove it.